testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestParquetArrayOfStructsColumn.java

```diff
@@ -0,0 +1,114 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.tests.product.hive;
+
+import com.google.inject.Inject;
+import io.trino.tempto.AfterTestWithContext;
+import io.trino.tempto.BeforeTestWithContext;
+import io.trino.tempto.hadoop.hdfs.HdfsClient;
+import org.testng.annotations.Test;
+
+import java.nio.file.Paths;
+
+import static io.trino.tempto.assertions.QueryAssert.Row.row;
+import static io.trino.tempto.assertions.QueryAssert.assertThat;
+import static io.trino.tests.product.TestGroups.PARQUET;
+import static io.trino.tests.product.TestGroups.STORAGE_FORMATS;
+import static io.trino.tests.product.utils.QueryExecutors.onTrino;
+
+public class TestParquetArrayOfStructsColumn
+        extends HiveProductTest
+{
+    @Inject
+    private HdfsClient hdfsClient;
+
+    @BeforeTestWithContext
+    public void setup()
+            throws Exception
+    {
+        hdfsClient.createDirectory("/user/hive/warehouse/TestParquetArrayOfStructsColumn/events");
+        saveOnHdfs(hdfsClient,
+                Paths.get("/docker/presto-product-tests/parquet/events.parquet"),
+                "/user/hive/warehouse/TestParquetArrayOfStructsRow/events/events.parquet");
+
+        hdfsClient.createDirectory("/user/hive/warehouse/TestParquetArrayOfStructsColumn/events-proto");
+        saveOnHdfs(hdfsClient,
+                Paths.get("/docker/presto-product-tests/parquet/events-proto-repeated-group.parquet"),
+                "/user/hive/warehouse/TestParquetArrayOfStructsRow/events-proto/events-proto-repeated-group.parquet");
+    }
+
+    @AfterTestWithContext
+    public void cleanup()
+    {
+        hdfsClient.delete("/user/hive/warehouse/TestParquetArrayOfStructsColumn");
+    }
+
+    @Test(groups = {PARQUET, STORAGE_FORMATS})
+    public void testSelectArrayOfStructsColumn()
+    {
+        /*
+        message custom_schema {
+          optional binary id (STRING);
+          optional group entries (LIST) {
+            repeated group list {
+              optional group element {
+                optional binary name (STRING);
+                optional group before {
+                  optional binary value (STRING); }
+                optional group after {
+                  optional binary value (STRING); }}}}}
+         */
+        onTrino().executeQuery("CREATE TABLE events (\n" +
+                "entries ARRAY(ROW(name VARCHAR, before ROW(value VARCHAR), after ROW(value VARCHAR))))\n" +
+                "WITH (\n" +
+                "external_location = 'hdfs:///user/hive/warehouse/TestParquetArrayOfStructsRow/events',\n" +
+                "format = 'PARQUET');");
+
+        assertThat(onTrino().executeQuery("SELECT entries FROM events"))
+                .containsExactlyInOrder(row(
```

Comment:
I think I have tried all possible ways to complete the assertion but it still fails:

```
tests               | 0 - expected: 
[{name=test-entry-0, before={value=before-0}, after={value=after-0}}, {name=test-entry-1, before={value=before-1}, after={value=after-1}}, {name=test-entry-2, before={value=before-2}, after={value=after-2}}]|
tests               | 0 - actual:   
[{name=test-entry-0, before={value=before-0}, after={value=after-0}}, {name=test-entry-1, before={value=before-1}, after={value=after-1}}, {name=test-entry-2, before={value=before-2}, after={value=after-2}}]|
```

Reply:
Leaving an array size assertion and a TODO.
Seems like implementing support of nested rows is required on Tempto side.